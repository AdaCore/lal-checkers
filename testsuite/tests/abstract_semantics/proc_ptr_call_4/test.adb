procedure Test is
   X : Integer;
   Y : Integer := 1;
   Z : Boolean;

   procedure A is
   begin
      X := 12;
   end A;

   procedure B(X : Boolean) is
      procedure C is
      begin
         X := 14;
      end C;

      P : access procedure := null;
   begin
      if X then
         P := A'Access;
      else
         P := C'Access;
      end if;
      P.all;
   end B;
begin
   B(Z);
end Test;
