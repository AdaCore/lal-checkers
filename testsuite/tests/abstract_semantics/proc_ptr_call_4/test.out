{
  "always": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "$stack": "({1: '[-2147483648, 2147483647]'}, 0)", 
          "Y": "[-2147483648, 2147483647]", 
          "Z": "{False, True}"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "$stack": "({1: '[-2147483648, 2147483647]'}, 0)", 
          "Y": "[1, 1]", 
          "Z": "{False, True}"
        }
      }
    ], 
    "assign2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "start0"
        ], 
        "values": {
          "$stack": "({1: '[-2147483648, 2147483647]'}, 0)", 
          "Y": "[1, 1]", 
          "Z": "{False, True}"
        }
      }
    ], 
    "assign3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "start0"
        ], 
        "values": {
          "$stack": "({1: '[-2147483648, 2147483647]'}, 0)", 
          "Y": "[1, 1]", 
          "Z": "{False, True}"
        }
      }
    ], 
    "assign4": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "start0"
        ], 
        "values": {
          "$stack": "({1: '[12, 14]'}, 0)", 
          "Y": "[1, 1]", 
          "Z": "{False, True}"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "$stack": "({}, 0)", 
          "Y": "[-2147483648, 2147483647]", 
          "Z": "{False, True}"
        }
      }
    ]
  }, 
  "le_t_eq_v": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "$stack": "({1: '[-2147483648, 2147483647]'}, 0)", 
          "Y": "[-2147483648, 2147483647]", 
          "Z": "{False, True}"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "$stack": "({1: '[-2147483648, 2147483647]'}, 0)", 
          "Y": "[1, 1]", 
          "Z": "{False, True}"
        }
      }
    ], 
    "assign2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "start0"
        ], 
        "values": {
          "$stack": "({1: '[-2147483648, 2147483647]'}, 0)", 
          "Y": "[1, 1]", 
          "Z": "{False, True}"
        }
      }
    ], 
    "assign3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "start0"
        ], 
        "values": {
          "$stack": "({1: '[-2147483648, 2147483647]'}, 0)", 
          "Y": "[1, 1]", 
          "Z": "{False, True}"
        }
      }
    ], 
    "assign4": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "start0"
        ], 
        "values": {
          "$stack": "({1: '[12, 14]'}, 0)", 
          "Y": "[1, 1]", 
          "Z": "{False, True}"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "$stack": "({}, 0)", 
          "Y": "[-2147483648, 2147483647]", 
          "Z": "{False, True}"
        }
      }
    ]
  }
}
